### PR TITLE
Fix assertion when quantize a cue point near the end

### PR DESCRIPTION
**Pull request by daschuer**
This fixes the false positive assertion:

```
Critical [Main]: DEBUG ASSERT: "cuePos <= total" in function double CueControl::quantizeCuePoint(double, CueControl::QuantizeMode) at /tmp/mixxx/src/engine/controls/cuecontrol.cpp:1641
```

### CODE REVIEW

File: src/engine/controls/cuecontrol.cpp [hunk 1]
```diff
@@ -1571,15 +1572,22 @@ CueControl::TrackAt CueControl::getTrackAt() const {
 
 double CueControl::getQuantizedCurrentPosition() {
     SampleOfTrack sampleOfTrack = getSampleOfTrack();
-    const double currentPos = sampleOfTrack.current;
+    double currentPos = sampleOfTrack.current;
     const double total = sampleOfTrack.total;
 
+    // Note: currentPos can be passed the end of the track, in the padded
```
Comment (uklotzde) [inline src/engine/controls/cuecontrol.cpp]: type: "past"

File: src/engine/controls/cuecontrol.cpp [hunk 2]
```diff
@@ -1571,15 +1572,22 @@ CueControl::TrackAt CueControl::getTrackAt() const {
...
+    // a future runs, depending on the buffering.
+    currentPos = math_min(currentPos, total);
+
     // Don't quantize if quantization is disabled.
     if (!m_pQuantizeEnabled->toBool()) {
         return currentPos;
     }
 
     double closestBeat = m_pClosestBeat->get();
+    // Note: closestBeat can be an interpolated beat passed the end of the track,
```
Comment (uklotzde) [inline src/engine/controls/cuecontrol.cpp]: typo: "past"

**Review by uklotzde — CHANGES_REQUESTED** [review]
Just some typos

**Comment by daschuer** [comment]
Done, Thank you for review.

File: src/engine/controls/cuecontrol.cpp [hunk 3]
```diff
@@ -1571,15 +1572,22 @@ CueControl::TrackAt CueControl::getTrackAt() const {
 
 double CueControl::getQuantizedCurrentPosition() {
     SampleOfTrack sampleOfTrack = getSampleOfTrack();
-    const double currentPos = sampleOfTrack.current;
+    double currentPos = sampleOfTrack.current;
     const double total = sampleOfTrack.total;
 
+    // Note: currentPos can be paset the end of the track, in the padded
```
Comment (Holzhaus) [inline src/engine/controls/cuecontrol.cpp]:
```suggestion
    // Note: currentPos can be past the end of the track, in the padded
```

**Comment by daschuer** [comment]
Ups, it was probably too late yesterday. I have amend the last faulty commit .

**Comment by uklotzde** [comment]
LGTM. Thanks for fixing this issue.

**Pull request merged**